pkg/controllers/job/plugins/distributed-framework/mpi/mpi.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2022 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mpi
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	batch "volcano.sh/apis/pkg/apis/batch/v1alpha1"
+	"volcano.sh/volcano/pkg/controllers/job/helpers"
+	pluginsinterface "volcano.sh/volcano/pkg/controllers/job/plugins/interface"
+	"volcano.sh/volcano/pkg/controllers/job/plugins/svc"
+)
+
+const (
+	MPIPluignName = "mpi"
+	defaultPort   = 22
+	defaultMaster = "master"
+	defaultWorker = "worker"
+	mpi_host      = "MPI_HOST"
+)
+
+type mpiPlugin struct {
+	mpiArguments []string
+	clientset    pluginsinterface.PluginClientset
+	masterName   string
+	workerName   string
+	port         int
+}
+
+// New creates mpi plugin.
+func New(client pluginsinterface.PluginClientset, arguments []string) pluginsinterface.PluginInterface {
+	mp := mpiPlugin{mpiArguments: arguments, clientset: client}
+	mp.addFlags()
+	return &mp
+}
+
+func NewInstance(arguments []string) mpiPlugin {
+	mp := mpiPlugin{mpiArguments: arguments}
+	mp.addFlags()
+	return mp
+}
+
+func (mp *mpiPlugin) addFlags() {
+	flagSet := flag.NewFlagSet(mp.Name(), flag.ContinueOnError)
+	flagSet.StringVar(&mp.masterName, "master", defaultMaster, "name of master role task")
+	flagSet.StringVar(&mp.workerName, "worker", defaultWorker, "name of worker role task")
+	flagSet.IntVar(&mp.port, "port", defaultPort, "open port for containers")
+	if err := flagSet.Parse(mp.mpiArguments); err != nil {
+		klog.Errorf("plugin %s flagset parse failed, err: %v", mp.Name(), err)
+	}
+}
+
+func (mp *mpiPlugin) Name() string {
+	return MPIPluignName
+}
+
+func (mp *mpiPlugin) OnPodCreate(pod *v1.Pod, job *batch.Job) error {
+
+	isMaster := false
+	workerHosts := ""
+	env := v1.EnvVar{}
+	if helpers.GetTaskKey(pod) == mp.masterName {
+		formateENVKey := strings.Replace(mp.workerName, "-", "_", -1)
+		key := fmt.Sprintf(svc.ConfigMapTaskHostFmt, formateENVKey)
+		workerHostsArr := strings.Split(svc.GenerateHosts(job)[key], "\n")
```

Comment:
l think it better to define another `GenerateWorkerHosts` in mpi plugin, it's not so good to depend on the code of svc plugin

Reply:
done